File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridTable/DataGridTable.cs
using KarveControls.DataGrid.DataGridHelpers.DataGridColumnVirtual;

namespace KarveControls.DataGrid.DataGridHelpers.DataGridTable
{
    public class DataGridTable
    {

        #region "   .   VARIABLES.  "

        public enum euCriteriosJoin
        {
            FromJoin,
            InnerJoin,
            LeftJoin,
            RightJoin,
            FullJoin,
            Union,
            UnionAll,
            FromFrom
        }

        int _Item;
        string _Table;
        string _Alias;
        euCriteriosJoin _Join;
        string _Criterio;
        DataGridTables _TablasVirtual = new DataGridTables();
        DataGridColumnVirtuales _ColumnasVirtual = new DataGridColumnVirtuales();

        private string _Name;
        #endregion

        #region "   -   PROPERTY.   "

        public int Item {
            get { return _Item; }
            set { _Item = value; }
        }

        public string Table {
            get { return _Table; }
            set { _Table = value; }
        }

        public string AliasTabla {
            get { return _Alias; }
            set { _Alias = value; }
        }

        public euCriteriosJoin Join {
            get { return _Join; }
            set { _Join = value; }
        }

        public string JoinText {
            get {
                string sJoin = "";
                switch (_Join) {
                    case euCriteriosJoin.FromJoin:
                        sJoin = " FROM ";
                        break;
                    case euCriteriosJoin.InnerJoin:
                        sJoin = " INNER JOIN ";
                        break;
                    case euCriteriosJoin.LeftJoin:
                        sJoin = " LEFT OUTER JOIN ";
                        break;
                    case euCriteriosJoin.RightJoin:
                        sJoin = " RIGHT OUTER JOIN ";
                        break;
                    case euCriteriosJoin.FromFrom:
                        sJoin = ", ";
                        break;
                    case euCriteriosJoin.Union:
                        sJoin = " UNION ";
                        break;
                    case euCriteriosJoin.UnionAll:
                        sJoin = " UNION ALL ";
                        break;
                    case euCriteriosJoin.FullJoin:
                        sJoin = " FULL OUTER JOIN ";
                        break;
                }
                return sJoin;
            }
        }


        public string Criterio {
            get { return _Criterio; }
            set { _Criterio = value; }
        }

        public DataGridTables TablasVirtual {
            get { return _TablasVirtual; }
            set { _TablasVirtual = value; }
        }

        public DataGridColumnVirtuales ColumnasTablaVirtual {
            get { return _ColumnasVirtual; }
            set { _ColumnasVirtual = value; }
        }

        public object Name {
            get {
                if (!string.IsNullOrEmpty(_Name)) {
                    return _Name;
                } else {
                    return _Table;
                }
            }
            set { _Name = (string)value; }
        }

        #endregion

    }
}

File: documentation/Karve/CustomControls/DataGrid/Clases/DataGridTable/DataGridTable.vb

Public Class DataGridTable

#Region "   .   VARIABLES.  "

    Enum euCriteriosJoin
        FromJoin
        InnerJoin
        LeftJoin
        RightJoin
        FullJoin
        Union
        UnionAll
        FromFrom
    End Enum

    Dim _Item As Integer
    Dim _Table As String
    Dim _Alias As String
    Dim _Join As euCriteriosJoin
    Dim _Criterio As String
    Dim _TablasVirtual As New DataGridTables
    Dim _ColumnasVirtual As New DataGridColumnVirtuales
    Private _Name As String

#End Region

#Region "   -   PROPERTY.   "

    Public Property Item As Integer
        Get
            Return _Item
        End Get
        Set(value As Integer)
            _Item = value
        End Set
    End Property

    Public Property Table As String
        Get
            Return _Table
        End Get
        Set(value As String)
            _Table = value
        End Set
    End Property

    Public Property AliasTabla As String
        Get
            Return _Alias
        End Get
        Set(value As String)
            _Alias = value
        End Set
    End Property

    Public Property Join As euCriteriosJoin
        Get
            Return _Join
        End Get
        Set(value As euCriteriosJoin)
            _Join = value
        End Set
    End Property

    Public ReadOnly Property JoinText As String
        Get
            Dim sJoin As String = ""
            Select Case _Join
                Case euCriteriosJoin.FromJoin : sJoin = " FROM "
                Case euCriteriosJoin.InnerJoin : sJoin = " INNER JOIN "
                Case euCriteriosJoin.LeftJoin : sJoin = " LEFT OUTER JOIN "
                Case euCriteriosJoin.RightJoin : sJoin = " RIGHT OUTER JOIN "
                Case euCriteriosJoin.FromFrom : sJoin = ", "
                Case euCriteriosJoin.Union : sJoin = " UNION "
                Case euCriteriosJoin.UnionAll : sJoin = " UNION ALL "
                Case euCriteriosJoin.FullJoin : sJoin = " FULL OUTER JOIN "
            End Select
            Return sJoin
        End Get

    End Property

    Public Property Criterio As String
        Get
            Return _Criterio
        End Get
        Set(value As String)
            _Criterio = value
        End Set
    End Property

    Public Property TablasVirtual As DataGridTables
        Get
            Return _TablasVirtual
        End Get
        Set(value As DataGridTables)
            _TablasVirtual = value
        End Set
    End Property

    Public Property ColumnasTablaVirtual As DataGridColumnVirtuales
        Get
            Return _ColumnasVirtual
        End Get
        Set(value As DataGridColumnVirtuales)
            _ColumnasVirtual = value
        End Set
    End Property

    Public Property Name
        Get
            If _Name <> "" Then
                Return _Name
            Else
                Return _Table
            End If
        End Get
        Set(value)
            _Name = value
        End Set
    End Property

#End Region

End Class
